### ISSUE TITLE
No Luxury Need (for WLTKD) for new Cities if all Luxuries available

### ISSUE THREAD

**Issue by UndeadDevel** [LoneGazebo/Community-Patch-DLL#5707]
_1. Mod version (i.e Date - 4/23):_
8/31

_2. Mod list (if using Vox Populi only, leave blank):_
List Enhancements for VP (v5)
Trade Opportunities for VP (v8.1)
Filters in Trade Route Overview (VP - v1.3)
Promotion Flags EUI (v1.1)
Capture Great People for VP (v1)

_3. Error description:_ 
I've noticed that newly acquired (founded and conquered alike) cities don't request resources for WLTKD if all luxuries are already available. I've had one city for 50 turns now and it still isn't requesting any even though other cities, when they run out of turns of WLTKD, request one of the resources that I already have for one turn (no WLTKD on that turn) and then enter WLTKD again (kinda weird, but I wouldn't necessarily call this a bug). That means I'm missing out on the extra yields from WLTKD for all the new cities.
Granted, in such a situation I don't really need them to win the game as I'll be dominating the map already anyway, but it's still a bug IMO.

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. Make sure you have enabled logging before experiencing an error! Go here to find out how: http://forums.civfanatics.com/showthread.php?t=487482):

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves):

3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

4. Screenshots (optional):

**Comment by LoneGazebo** [comment]
Not a bug.

**Comment by UndeadDevel** [comment]
It is a bit inconsistent, though, as the other cities will still get a new WLTKD once theirs runs out (after coming up with a new desire of already available resource for one turn). This can be remedied somewhat by artificially creating a situation where one doesn't have all resources anymore (e.g. by trading all of one type away) but what is more likely to happen (and did in my game) is that the player won't realize this until many turns have passed, during which the player will wait around for the new cities to finally desire resources like the others...the way it currently works is simply unintuitive and would, IMO, be improved if new cities acted like existing cities.

**Comment by LoneGazebo** [comment]
With no WLTKD resources to trigger, though, you'd just be in eternal WLTKD.

**Comment by UndeadDevel** [comment]
Yes, that is what's happening with old cities; they run out of WLTKD and the desire an existing resource for one turn and then go back to WLTKD...it should be consistent across all cities, i.e. either no city desires a new resource if all are available or all do, new and old alike.

**Comment by LoneGazebo** [comment]
Ah, I see why, GetResourceDemanded is never properly reset. I'll fix. No WLTKD for free if you have all resources.
